nvm.sh

```diff
@@ -1828,6 +1828,17 @@ nvm() {
         NVM_PREFIX="$(nvm_node_prefix)"
         NVM_SUCCESS_MSG="Uninstalled node $VERSION"
       fi
+
+      # Check version dir permissions
+      local INVALID_FILES="$(find "$VERSION_PATH" -exec [ ! -w "{}" ] \; -exec echo "{}" \;)"
+      if [ "$INVALID_FILES" ]; then
```

Comment:
`$INVALID_FILES` contains a list of any files that we **don't** have write permissions for, so it's just a truthy test. If everything's good it should be an empty variable (falsey).

Works fine as is with sh, bash, zsh and dash but if you think it would help readability I could add in `-z`.

Maybe, if it's readability you're concerned about, it would be better to rename the var to `$FILES_WITHOUT_WRITE_PERMISIONS`. Then we could have:
```shell
if [ "$FILES_WITHOUT_WRITE_PERMISIONS" ]; then
  # Error message
fi
```
which I think makes it obvious what's going on. Or is that too verbose?